### PR TITLE
refactor(plugin-image): Refactor Pixabay Search component 

### PR DESCRIPTION
**Pull request by shn-srl**
- Combine Pixabay Search Button and modal into one self contained component
- Add white background to image plugin
- Remove white shadow above image plugin toolbar

### CODE REVIEW

File: packages/editor/src/plugins/image/components/pixabay-search-button.tsx [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  const isGermanLocale = lang === 'de'
+  const showTags = isGermanLocale && images.length === 0 && !isSearching
+
+  return (
+    <>
+      <button
+        data-qa="plugin-image-pixabay-search-button"
+        onClick={() => setIsOpen(true)}
+        onFocus={() => onFocus()}
+        onBlur={() => onBlur()}
```
Comment (elbotho) [inline packages/editor/src/plugins/image/components/pixabay-search-button.tsx]:
```suggestion
        onFocus={onFocus}
        onBlur={onBlur}
```

File: packages/editor/src/plugins/image/components/pixabay-search-button.tsx [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        className="mb-4 flex min-w-full flex-shrink-0 items-center justify-center rounded-lg bg-editor-primary-200 p-1 py-2 font-semibold text-almost-black text-gray-800 hover:bg-editor-primary-300"
```
Comment (elbotho) [inline packages/editor/src/plugins/image/components/pixabay-search-button.tsx]:
```suggestion
        className="serlo-button-editor-primary mb-4 min-w-full rounded-lg px-1 py-2 font-semibold"
```

File: packages/editor/src/plugins/image/components/pixabay-search-button.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+              type="text"
+              value={query}
+              onChange={handleInputChange}
+              placeholder={imageStrings.search}
+              onKeyDown={handleKeyDown}
+              className="w-full rounded-lg border-0 bg-yellow-100 py-2 pl-4 pr-10 text-gray-600"
+            />
+            {query && (
+              <div
+                className="absolute bottom-0 right-0 top-0 flex cursor-pointer items-center justify-center px-4 text-gray-400"
```
Comment (elbotho) [inline packages/editor/src/plugins/image/components/pixabay-search-button.tsx]:
```suggestion
                className="absolute bottom-1 right-1 top-1 flex items-center p-2 text-gray-400 hover:text-black"
```

File: packages/editor/src/plugins/image/components/pixabay-search-button.tsx [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+              ref={inputRef}
+              type="text"
+              value={query}
+              onChange={handleInputChange}
+              placeholder={imageStrings.search}
+              onKeyDown={handleKeyDown}
+              className="w-full rounded-lg border-0 bg-yellow-100 py-2 pl-4 pr-10 text-gray-600"
+            />
+            {query && (
+              <div
```
Comment (elbotho) [inline packages/editor/src/plugins/image/components/pixabay-search-button.tsx]: this should be a `<button>`.

File: packages/editor/src/plugins/image/components/pixabay-search-button.tsx [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+                  </button>
+                ))}
+              </div>
+            </div>
+          )}
+
+          <div
+            className={cn(
+              'max-h-[500px]',
+              'mt-4 flex flex-wrap px-8',
```
Comment (elbotho) [inline packages/editor/src/plugins/image/components/pixabay-search-button.tsx]:
```suggestion
              'mt-4 flex flex-wrap pl-8 pr-10',
```
(makes the right side of the images line up with the search input

**Review by elbotho — APPROVED** [review]
I suggested some improvements but I realize that his is not the real scope of this PR so I'll approve right now.
Still I'm happy if you add the changes :)

File: packages/editor/src/plugins/image/components/pixabay-search-button.tsx (hunk 4 above)
Comment (shn-srl) [reply packages/editor/src/plugins/image/components/pixabay-search-button.tsx]: Done :heavy_check_mark:

**Pull request merged**